networkx/drawing/layout.py

```diff
@@ -617,3 +619,178 @@ def flatten(l):
+        eu_dist = math.sqrt(diff_x ** 2.0 + diff_y ** 2.0)
+        eu_dist_cubed = eu_dist ** 3.0
+        edx = kappa * (diff_x - (la * diff_x / eu_dist))
+        edy = kappa * (diff_y - (la * diff_y / eu_dist))
+        edxx = kappa * (1 - (la * (diff_y ** 2.0)) / eu_dist_cubed)
+        edyy = kappa * (1 - (la * (diff_x ** 2.0)) / eu_dist_cubed)
+        edxy = (kappa * diff_y * la * diff_x) / eu_dist_cubed
+        return edx, edy, edxx, edyy, edxy
+
+    # Initialize delta and compute partial derivatives for every node
+    delta = np.zeros((pos_arr.shape[0], 1), dtype=dist.dtype)
+    derivatives = np.zeros((pos_arr.shape[0], number_of_derivatives),
+                           dtype=dist.dtype)
+    partial_derivatives = np.vectorize(compute_derivatives)
+    for u in range(n):
+        elements = np.arange(n)
+        elements = elements[elements != u]
+        derivatives_m = np.sum(np.array(list(
+            partial_derivatives(pos_arr[u][0], pos_arr[u][1], pos_arr[elements, 0],
+                                pos_arr[elements, 1], k[u, elements],
+                                l[u, elements])),
+            dist.dtype).reshape(number_of_derivatives, n - 1), axis=1)
+        derivatives[u] = derivatives_m
+        delta[u] = math.sqrt(derivatives_m[0] ** 2.0 + derivatives_m[1] ** 2.0)
+
+    # Select node with max delta
+    m = np.argmax(delta)
+
+    outer_iter = 0
+    while delta[m] > D and (outer_iter < iterations):
+        outer_iter += 1
+        inner_iter = 0
+
+        elements = np.arange(n)
+        elements = elements[elements != m]
+
+        while delta[m] > D and (inner_iter < tolerance):
+            inner_iter += 1
+            old_px = pos_arr[m, 0]
+            old_py = pos_arr[m, 1]
+            dx = (derivatives[m, 4] * derivatives[m, 1] - derivatives[m, 3]
+                  * derivatives[m, 0]) / (derivatives[m, 2] * derivatives[m, 3]
+                                          - derivatives[m, 4] ** 2.0)
+            dy = (derivatives[m, 2] * derivatives[m, 1] - derivatives[m, 4]
+                  * derivatives[m, 0]) / (derivatives[m, 4] ** 2.0 -
+                                          derivatives[m, 2] * derivatives[m, 3])
+
+            # Move node m to a new position
+            pos_arr[m, 0] += dx
+            pos_arr[m, 1] += dy
+
+            # Compute new partial derivatives of node m
+            derivatives_m = np.sum(np.array(list(
+                partial_derivatives(pos_arr[m, 0], pos_arr[m, 1], pos_arr[elements, 0],
+                                    pos_arr[elements, 1], k[m, elements],
+                                    l[m, elements])),
+                                    dist.dtype).reshape(
+                                    number_of_derivatives, n - 1), axis=1)
+            derivatives[m] = derivatives_m
+            delta[m] = math.sqrt(derivatives_m[0] ** 2.0 + derivatives_m[1] ** 2.0)
+
+        # Compute new partial derivatives for every node
+        for u in range(n):
```

Comment:
This loop, and the one on line 730, would be clearer as a list comprehension, something like `delta = np.array([update_derivative(...) for d in delta])`. (For this loop, you would then write a new function `update_derivative` that would perform the update.)